### PR TITLE
Pluggable store

### PR DESCRIPTION
**Pull request by neekolas**
## Problems
- The Waku node has a concrete reference to the `go-waku` store in it's struct instead of an interface
- The Waku node creates a new store instance as part of the `Start()` function

These two issues make it impossible to inject a new type of store into the existing node, so our wrapper code would need to completely recreate the `waku/v2/node` package in order to make any changes to the store.

## Solution
- Create a new field on the `WakuNode` struct to store a `storeFactory` constructor function. This is set as a default function that recreates the existing behavior if it is not overwritten. The `storeFactory` takes a `WakuNode` as it's only argument, as that is needed to fulfill the existing store creation behavior (and likely ours as well)
- Create a new parameter and helper function that allows for callers to manually set a `storeFactory`
- Create a `MessageChannel` function on the new `Store` instance that returns a reference to the channel, instead of accessing it as an exported method from the struct.

This solution still creates the store inside the `Start()` function, but now allows for the logic to be customized. I think this is a reasonable compromise that should be workable if we want to PR it into Status' main branch.

### CODE REVIEW

File: waku/v2/discv5/discover.go [hunk 1]
```diff
@@ -318,7 +318,8 @@ func evaluateNode(node *enode.Node) bool {
 		return false
 	}
 
-	if !isWakuNode(node) || !hasTCPPort(node) {
+	//  TODO: consider node filtering based on ENR; we do not filter based on ENR in the first waku discv5 beta stage
+	if /*!isWakuNode(node) ||*/ !hasTCPPort(node) {
```
Comment (neekolas) [inline waku/v2/discv5/discover.go:322]: I must have picked this up from `status/master` along the way. I didn't make this change.

File: waku/v2/node/wakunode2.go [hunk 2]
```diff
@@ -433,6 +447,10 @@ func (w *WakuNode) mountDiscV5() error {
 		discv5.WithAutoUpdate(w.opts.discV5autoUpdate),
 	}
 
+	if w.opts.advertiseAddr != nil {
+		discV5Options = append(discV5Options, discv5.WithAdvertiseAddr(*w.opts.advertiseAddr))
```
Comment (neekolas) [inline waku/v2/node/wakunode2.go:451]: This change also came from `status-im/go-waku:master`

File: waku/v2/protocol/store/waku_store.go [hunk 3]
```diff
@@ -240,6 +240,15 @@ type WakuStore struct {
 	swap         *swap.WakuSwap
 }
 
+type Store interface {
```
Comment (neekolas) [inline waku/v2/protocol/store/waku_store.go:243]: This covers all exported methods of the existing Store struct

**Review by mkobetic — APPROVED** [review]
Yup, this looks pretty good to me.

**Pull request merged**